### PR TITLE
[IND-511] e2e test for limit order matching

### PR DESCRIPTION
**Pull request by dydxwill**
### Changelist
e2e test that places:
- 1 sell order
- 1 buy order
- checks that fill occurred
- checks downstream dependencies (websockets, Postgres, orderbook in Redis, API, candles)

### Test Plan
ran locally

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

### CODE REVIEW

File: e2e-testing/README.md [hunk 1]
```diff
@@ -16,3 +16,33 @@ In another terminal, run
 ```
 pnpm build && pnpm test
 ```
+
+#### Quickest way to reset the network/clear all Indexer data sources without rebuilding from scratch
```
Comment (Christopher-Li) [inline e2e-testing/README.md:20]: maybe as a followup we can file a task to write a script to do all of this? Is the network/indexer reset in the e2e tests right now?

File: e2e-testing/__tests__/orders.test.ts [hunk 2]
```diff
@@ -0,0 +1,387 @@
...
+
+type OrderDetails = {
+  mnemonic: string;
+  timeInForce: number;
+  orderFlags: number;
+  side: number;
+  clobPairId: number;
+  quantums: number;
+  subticks: number;
+};
```
Comment (Christopher-Li) [inline e2e-testing/__tests__/orders.test.ts:59]: nit: I think some of these can be in `constants.ts`

File: e2e-testing/__tests__/orders.test.ts [hunk 3]
```diff
@@ -0,0 +1,387 @@
...
+  it('test orders', async () => {
+    const indexerClient = new IndexerClient(Network.local().indexerConfig);
+    const heightResp: HeightResponse = await indexerClient.utility.getHeight();
+    const height: number = heightResp.height;
+    connectAndValidateSocketClient();
+
+    // place all orders
+    for (const order of orderDetails) {
+      const modifiedOrder: IPlaceOrder = defaultOrder;
+      modifiedOrder.clientId = Math.floor(Math.random() * 1000000000);
```
Comment (Christopher-Li) [inline e2e-testing/__tests__/orders.test.ts:119]: nit: move into a helper so it's obvious what this is doing

File: e2e-testing/__tests__/orders.test.ts [hunk 4]
```diff
@@ -0,0 +1,387 @@
...
+        close: '50000',
+        baseTokenVolume: '0.0005',
+        usdVolume: '25',
+        trades: 1,
+      }),
+    );
+  });
+
+
+  function connectAndValidateSocketClient(): void {
```
Comment (Christopher-Li) [inline e2e-testing/__tests__/orders.test.ts:330]: is there a way to make this socket logic more generic between orders and transfer e2e tests?

File: e2e-testing/__tests__/orders.test.ts [hunk 5]
```diff
@@ -0,0 +1,387 @@
...
+    const mySocket = new SocketClient(
+      Network.local().indexerConfig,
+      () => {
+        console.log('open');
```
Comment (Christopher-Li) [inline e2e-testing/__tests__/orders.test.ts:334]: nit: should we remove the console.log statements?

File: e2e-testing/README.md [hunk 6]
```diff
@@ -16,3 +16,33 @@ In another terminal, run
 ```
```
Comment (Christopher-Li) [inline e2e-testing/README.md:16]: nit: Did you add a github action to run these e2e tests or a github action to at least check that these files pass build and lint tests?

File: e2e-testing/README.md (hunk 1 above)
Comment (dydxwill) [reply e2e-testing/README.md:20]: created https://linear.app/dydx/issue/IND-551/write-script-for-resetting-the-networkclearing-all-indexer-data to track

File: e2e-testing/__tests__/orders.test.ts (hunk 2 above)
Comment (dydxwill) [reply e2e-testing/__tests__/orders.test.ts:59]: moved

File: e2e-testing/__tests__/orders.test.ts (hunk 3 above)
Comment (dydxwill) [reply e2e-testing/__tests__/orders.test.ts:119]: done

File: e2e-testing/__tests__/orders.test.ts (hunk 4 above)
Comment (dydxwill) [reply e2e-testing/__tests__/orders.test.ts:330]: done

File: e2e-testing/__tests__/orders.test.ts (hunk 5 above)
Comment (dydxwill) [reply e2e-testing/__tests__/orders.test.ts:334]: done

File: e2e-testing/README.md (hunk 6 above)
Comment (dydxwill) [reply e2e-testing/README.md:16]: will do this separately. Want to make sure it's not flaky in github actions before submitting. created ticket here: https://linear.app/dydx/issue/IND-550/github-actions-for-buildingrunning-e2e-test

File: e2e-testing/clear-all-kafa-topics.sh [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+# Topics to process
+declare -a topics=("to-ender" "to-vulcan" "to-websockets-orderbooks" "to-websockets-subaccounts" "to-websockets-trades" "to-websockets-markets" "to-websockets-candles")
+
+for topic in "${topics[@]}"
+do
+    echo "Deleting topic: $topic"
+    kafka-topics.sh --zookeeper $ZOOKEEPER --delete --topic $topic
+    echo "Creating topic: $topic"
+    kafka-topics.sh --create --zookeeper $ZOOKEEPER --replication-factor 1 --partitions 1 --topic $topic
```
[the remark this answers is not in the document]
Comment (Christopher-Li) [reply e2e-testing/clear-all-kafa-topics.sh:14]: nit: seems like a good edit

**Review by Christopher-Li — APPROVED** [review]

**Pull request merged**